pkgs/by-name/up/upbound/package.nix

```diff
@@ -1,32 +1,83 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{
+  system ? builtins.currentSystem,
+  lib,
+  fetchurl,
+  installShellFiles,
+  versionCheckHook,
+  stdenvNoCC,
+  version-channel ? "stable",
+}:
+let
+  sources =
+    if "${version-channel}" == "main" then import ./sources-main.nix else import ./sources-stable.nix;
+  arch = sources.archMap.${system};
 
-buildGoModule rec {
-  pname = "upbound";
-  version = "0.34.0";
+in
+stdenvNoCC.mkDerivation {
+  pname = if "${version-channel}" == "main" then "upbound-main" else "upbound";
+  version = sources.version;
+  srcs = [
+    (fetchurl {
+      url = sources.fetchurlAttrSet.${system}.docker-credential-up.url;
+      sha256 = sources.fetchurlAttrSet.${system}.docker-credential-up.hash;
+    })
 
-  src = fetchFromGitHub {
-    owner = pname;
-    repo = "up";
-    rev = "v${version}";
-    hash = "sha256-npWCGR0/iXAF7efreWhpwqboXJ24lqk5TPaB7glwEFI=";
-  };
+    (fetchurl {
+      url = sources.fetchurlAttrSet.${system}.up.url;
+      sha256 = sources.fetchurlAttrSet.${system}.up.hash;
+    })
+  ];
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    cp ./${arch}/up $out/bin/up
+    chmod +x $out/bin/up
+
+    cp ./${arch}/docker-credential-up $out/bin/docker-credential-up
+    chmod +x $out/bin/docker-credential-up
 
-  vendorHash = "sha256-f/RPyjrC5FsOjC5wnBtqEbLeVX7VeiVYFpRXcQ2lTc4=";
+    runHook postInstall
+  '';
 
-  subPackages = [ "cmd/docker-credential-up" "cmd/up" ];
+  postInstall = ''
+    installShellCompletion --bash --name up <(echo complete -C up up)
+  '';
 
-  ldflags = [
-    "-s"
-    "-w"
-    "-X github.com/upbound/up/internal/version.version=v${version}"
+  versionCheckProgramArg = "version";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
   ];
 
-  meta = with lib; {
-    description =
-      "CLI for interacting with Upbound Cloud, Upbound Enterprise, and Universal Crossplane (UXP)";
+  system = system;
+
+  doCheck = !(with stdenvNoCC; isDarwin && isAarch64); # turning off tests in this case as ofBorg does not ternimate waiting for a builder
+
+  passthru.updateScript = [
+    ./update.sh
+    "${version-channel}"
+  ];
+
+  passthru.tests = {
+    versionCheck = versionCheckHook;
+  };
+
+  meta = {
+    description = "CLI for interacting with Upbound Cloud, Upbound Enterprise, and Universal Crossplane (UXP)";
     homepage = "https://upbound.io";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ lucperkins ];
+    license = lib.licenses.asl20;
```

Comment:
This needs to be an `unfree` license

Reply:
Why was this marked as resolved? This needs to be an unfree license. As far as I can tell, the source repo for Upbound is no longer public.